src/libs/vehicle/ardupilot/ardupilot.ts

```diff
@@ -443,6 +501,14 @@ export abstract class ArduPilotVehicle<Modes> extends Vehicle.AbstractVehicle<Mo
     return this._isArmed
   }
 
+  /**
+   * Check if the UI should show the takeoff button
+   * @returns {boolean}
+   */
+  showTakeoff(): boolean {
+    return !this._flying
+  }
+
```

Comment:
I don't like this coupling to the UI here. I think the ardupilot and vehicle modules should be agnostic to the UI.

But this is an easy change. We can just call the function `flying()`, and the UI decides that if the vehicle is flying, it should show "Land" and not "Takeoff".